modules/mbedtls/Kconfig.tls-generic

```diff
@@ -247,6 +247,7 @@ config MBEDTLS_SOME_CIPHER_ENABLED
 
 config MBEDTLS_CIPHER_AES_ENABLED
 	bool "AES block cipher"
+	select MBEDTLS_CIPHER
```

Comment:
Agreed with Valerio. I think you'll want to separately enable `CONFIG_MBEDTLS_CIPHER` for when you want to use the cipher layer (`mbedtls_cipher_*()`).

Reply:
That is true, but it is weird to me that one does: `CONFIG_MBEDTLS_CIPHER_AES_ENABLED=y` and is not able to do:

```
      cipher_info = mbedtls_cipher_info_from_type(MBEDTLS_CIPHER_AES_256_ECB);
```

The symbol has the same namespace .... Would make sense to me if it was `MBEDTLS_AES_ENABLED`.

I think it is misleading / inconsistent, but that's said it will bring more code so I understand if you still think it shouldn't be changed. Just let me know if that is what you think guys and I remove this commit from the pr.

Reply:
> The symbol has the same namespace .... Would make sense to me if it was MBEDTLS_AES_ENABLED.

You're right actually, and I agree that we should rename those Kconfig options. They actually end up as Mbed TLS defines without `CIPHER` nor `ENABLED` in their name.
We've been striving to map the Kconfig option names to their Mbed TLS define counterparts. And I see in `config-tls-generic.h` we have `MBEDTLS_AES_C` etc. so the Kconfig options should become `MBEDTLS_<name of the cipher>[_C]` (optional trailing `_C`, I am not decided on that one).

I think we'll still want to allow enabling individual ciphers without enabling the `cipher` layer. Or maybe what could be done here instead is to `imply` instead of `select` `MBEDTLS_CIPHER`. This would still allow disabling it. Though lately we've tried to move away from default-enabled options that are not strictly necessary here, so I'm not sure. Even if `mbedtls_cipher_*()` functions are not used they might cause some overhead just become some more code is compiled in in some other functions due to that. I haven't checked but that's at least what we've seen with PSA Crypto.